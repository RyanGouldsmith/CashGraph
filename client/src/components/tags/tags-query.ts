import gql from 'graphql-tag';

export const GetTagsQuery = gql`
  query GetTags($userId: String!) {
    tags(userId: $userId) {
      id
      name
      colour
    }
  }
`;

gql`
  type TagInput {
    name: TagName
    colour: TagColour
  }
`;

gql`
  enum TagName {
    HOLIDAY
    FOOD
    TRAVEL
    ENTERTAINMENT
  }
`;

gql`
  enum TagColour {
    GREEN
    RED
    BLUE
  }
`;
