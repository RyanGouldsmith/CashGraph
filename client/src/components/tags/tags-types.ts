export interface TagsQuery {
  tags: Array<TagType>;
}

export interface TagType {
  id: string;
  name: TagValues;
  colour: TagColours;
}

export enum TagValues {
  HOLIDAY,
  FOOD,
  TRAVEL,
  ENTERTAINMENT
}

export enum TagColours {
  GREEN,
  RED,
  BLUE
}
