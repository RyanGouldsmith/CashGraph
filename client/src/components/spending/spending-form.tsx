import React from 'react';
import { MutationFn } from 'react-apollo';

import { TagType } from '../tags/tags-types';

interface SpendingFormProps {
  submitForm: Function;
  spendingCallback: MutationFn;
  userId: String;
  titleRef: React.Ref<HTMLInputElement>;
  priceRef: React.Ref<HTMLInputElement>;
  tagRef: React.Ref<HTMLSelectElement>;
  spendingData: {
    title?: string;
    price?: number;
    tags?: Array<TagType>;
    selectedTag?: string;
  };
}

export const SpendingForm: React.SFC<SpendingFormProps> = ({
  submitForm,
  spendingCallback,
  userId,
  titleRef,
  priceRef,
  tagRef,
  spendingData: { title = '', price = 0, tags = [], selectedTag = '' }
}) => (
  <form onSubmit={submitForm(spendingCallback, userId)}>
    <label htmlFor="title">Enter the item you purchased</label>
    <input
      id="title"
      placeholder="Shoes"
      type="text"
      ref={titleRef}
      defaultValue={title}
    />
    <label htmlFor="price">The price of the item</label>
    <input
      id="price"
      type="number"
      min="0"
      ref={priceRef}
      step="0.01"
      defaultValue={price.toFixed(2)}
    />
    <label htmlFor="tag">Type of Tag</label>
    <select ref={tagRef} id="tag">
      {tags.map(tag => (
        <option value={tag.id} selected={tag.id === selectedTag}>
          {tag.name}
        </option>
      ))}
    </select>
    <button type="submit">submit</button>
  </form>
);
