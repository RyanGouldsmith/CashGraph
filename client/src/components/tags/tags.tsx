import React from 'react';
import { Query } from 'react-apollo';

import { GetTagsQuery } from './tags-query';
import { TagType } from './tags-types';
import { Loading } from '../loading/loading';

export const Tags: React.SFC<{}> = _ => {
  return (
    <React.Fragment>
      <h1>The tags are: </h1>
      <Query query={GetTagsQuery}>
        {({ loading, data }) => {
          if (loading) return <Loading />;
          if (data) {
            const { tags } = data;
            return (
              <div className="tags">
                {tags.map((tag: TagType) => {
                  return (
                    <p style={{ color: `${tag.colour}` }} className="tags__tag" key={tag.name}>
                      {tag.name}
                    </p>
                  );
                })}
              </div>
            );
          }
          return null;
        }}
      </Query>
    </React.Fragment>
  );
};
export default Tags;
