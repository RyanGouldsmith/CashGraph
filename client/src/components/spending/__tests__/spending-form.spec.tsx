import 'babel-polyfill';
import React from 'react';
import { render } from 'react-testing-library';
import { MutationFn } from 'react-apollo';

import { TagValues, TagColours } from '../../tags/tags-types';

import { SpendingForm } from '../spending-form';

test('match the snapshot for the data passed into the spending form', async () => {
  const spendingCallback: MutationFn = () => new Promise(resolve => resolve());

  const props = {
    submitForm: () => {},
    spendingCallback,
    userId: '123',
    titleRef: React.createRef<HTMLInputElement>(),
    priceRef: React.createRef<HTMLInputElement>(),
    tagRef: React.createRef<HTMLSelectElement>(),
    spendingData: {
      title: 'knownTitle',
      price: 10.0,
      tags: [
        {
          id: '123',
          name: TagValues.ENTERTAINMENT,
          colour: TagColours.BLUE
        },
        {
          id: '2345',
          name: TagValues.FOOD,
          colour: TagColours.RED
        }
      ],
      selectedTag: '123'
    }
  };

  expect(render(<SpendingForm {...props} />)).toMatchSnapshot();
});
