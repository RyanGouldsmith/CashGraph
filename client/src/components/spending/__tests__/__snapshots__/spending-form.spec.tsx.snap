// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match the snapshot for the data passed into the spending form 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form>
        <label
          for="title"
        >
          Enter the item you purchased
        </label>
        <input
          id="title"
          placeholder="Shoes"
          type="text"
          value="knownTitle"
        />
        <label
          for="price"
        >
          The price of the item
        </label>
        <input
          id="price"
          min="0"
          step="0.01"
          type="number"
          value="10.00"
        />
        <label
          for="tag"
        >
          Type of Tag
        </label>
        <select
          id="tag"
        >
          <option
            value="123"
          >
            3
          </option>
          <option
            value="2345"
          >
            1
          </option>
        </select>
        <button
          type="submit"
        >
          submit
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form>
      <label
        for="title"
      >
        Enter the item you purchased
      </label>
      <input
        id="title"
        placeholder="Shoes"
        type="text"
        value="knownTitle"
      />
      <label
        for="price"
      >
        The price of the item
      </label>
      <input
        id="price"
        min="0"
        step="0.01"
        type="number"
        value="10.00"
      />
      <label
        for="tag"
      >
        Type of Tag
      </label>
      <select
        id="tag"
      >
        <option
          value="123"
        >
          3
        </option>
        <option
          value="2345"
        >
          1
        </option>
      </select>
      <button
        type="submit"
      >
        submit
      </button>
    </form>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
